### ISSUE TITLE
api for design cells and library cells iteration

### ISSUE THREAD

**Issue by ahmednofal** [RsynTeam/rsyn-x#10]
There is not any way to iterate over the cells in the design through an Rsyn::Design or Rsyn::PhysicalDesign object. Also there is not any way to iterate over the libraryCells as well.
Should not they get added ?

**Comment by gaflach** [comment]
Hi, Ahmad.

Cells (more generically instances) are managed by a Rsyn::Module.
Library cells are managed by the Rsyn::Library.

```
Rsyn::Session session;

Rsyn::Module module = session.getTopModule();
for (Rsyn::Instance instance : module.allInstances()) { 
   // traverse all instances in the design 
}

Rsyn::Library library = session.getLibrary();
for (Rsyn::LibraryCell lcell = library.allLibraryCells()) { 
  // traverse all library cells in the library 
}
```

Instances represent cells, ports and modules (hierarchical designs).
Very often you may want to skip non-cell instances.

```
for (Rsyn::Instance instance : module.allInstances()) {
  if (instance.getType() != Rsyn::CELL)
    continue;
  Rsyn::Cell cell = instance.asCell();
}
```

Regards,
Guilherme Flach

**Comment by ahmednofal** [comment]
Yes I just found the allLibraryCells api thank you very much. But can not there be any wrapper function such as the one mentioned in 
x/src/x/opto/ufrgs/qpdp/IncrementalTimingDrivenQP.cpp
764://	for ( Rsyn::Cell cell : design.allCells() ) 

but commented out ? to handle the checking if the instance is a cell internally ?

**Comment by gaflach** [comment]
Yes, that could be added...
I'll add that.

**Comment by eder-matheus** [comment]
Hi, Ahmed.

Fixed #`173b91f`
